x-pack/plugins/security/public/management/api_keys/api_keys_grid/api_key_flyout.tsx

```diff
@@ -50,45 +58,103 @@ export interface ApiKeyFormValues {
   metadata: string;
 }
 
-export interface CreateApiKeyFlyoutProps {
+export interface ApiKeyFlyoutProps {
   defaultValues?: ApiKeyFormValues;
-  onSuccess?: (apiKey: CreateApiKeyResponse) => void;
+  onSuccess?: (
+    createApiKeyResponse: CreateApiKeyResponse | undefined,
+    updateApiKeyResponse: UpdateApiKeyResponse | undefined
+  ) => void;
   onCancel: FormFlyoutProps['onCancel'];
+  selectedApiKey?: ApiKey;
+  readonly?: boolean;
 }
 
 const defaultDefaultValues: ApiKeyFormValues = {
   name: '',
-  expiration: '',
   customExpiration: false,
-  customPrivileges: false,
+  expiration: '',
   includeMetadata: false,
-  role_descriptors: '{}',
   metadata: '{}',
+  customPrivileges: false,
+  role_descriptors: '{}',
 };
 
-export const CreateApiKeyFlyout: FunctionComponent<CreateApiKeyFlyoutProps> = ({
+export const ApiKeyFlyout: FunctionComponent<ApiKeyFlyoutProps> = ({
   onSuccess,
   onCancel,
   defaultValues = defaultDefaultValues,
+  selectedApiKey,
+  readonly = false,
 }) => {
-  const { services } = useKibana();
+  let formTitle = 'Create API Key';
+  let inProgressButtonText = 'Creating API Key…';
+  let errorTitle = 'create API key';
+
   const { value: currentUser, loading: isLoadingCurrentUser } = useCurrentUser();
+
+  let canEditApiKey = false;
+
+  if (selectedApiKey) {
+    // Collect data from the selected API key to pre-populate the form
+    const doesMetadataExist = Object.keys(selectedApiKey.metadata).length > 0;
+    const doCustomPrivilegesExist = Object.keys(selectedApiKey.role_descriptors ?? 0).length > 0;
+
+    defaultValues = {
```

Comment:
Great point! Ive made the change in the latest commit so that caller can pass in an `apiKey`, if not, the default values will be set/used by the flyout